### ISSUE TITLE
Fix S3456 FN: Support utf-8 literals

### ISSUE THREAD

**Issue by cristian-ambrosini-sonarsource** [SonarSource/sonar-dotnet#6392]
### Description

FN for S3456 (RedundantToCharArrayCall).

Using C#11 utf-8 string literals, the rule is not raising an issue.

### Repro steps

```csharp
    public class RedundantToCharArrayCall
    {
        public void MyMethod(int propertyValue)
        {
            var c = "some string"u8.ToArray()[10];        // FN     
            foreach (var v in "some string"u8.ToArray())  // FN
            {
                // ...
            }
        }
    }
```

### Expected behavior

The rule should raise an issue.

### Actual behavior

The rule does not raise an issue.
